homeassistant/components/fortios/device_tracker.py

```diff
@@ -0,0 +1,111 @@
+"""
+Support to use FortiOS device like FortiGate as device tracker.
+
+This component is part of the device_tracker platform.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/fortiosapi/
+"""
+import logging
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.device_tracker import (
+    DOMAIN, PLATFORM_SCHEMA, DeviceScanner)
+from homeassistant.const import CONF_HOST, CONF_TOKEN
+from homeassistant.const import CONF_VERIFY_SSL
+
+REQUIREMENTS = ['fortiosapi==0.10.8']
+
+_LOGGER = logging.getLogger(__name__)
+DEFAULT_VERIFY_SSL = False
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_TOKEN): cv.string,
+    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): cv.boolean
+})
+
+
+def get_scanner(hass, config):
+    """Validate the configuration and return a FortiOSDeviceScanner."""
+    from fortiosapi import FortiOSAPI
+
+    host = config[DOMAIN].get(CONF_HOST)
+    verify_ssl = config[DOMAIN].get(CONF_VERIFY_SSL)
+    token = config[DOMAIN].get(CONF_TOKEN)
+
+    _LOGGER.debug("host : %s", host)
+    _LOGGER.debug("verify_ssl : %s", verify_ssl)
+    _LOGGER.debug("token : %s", token)
+
+    _LOGGER.debug('fortios, get_scanner')
+
+    fgt = FortiOSAPI()
+
+    try:
+        fgt.tokenlogin(host, token, verify_ssl)
+    except Exception as ex:  # pylint: disable=broad-except
```

Comment:
We should catch a specific expected exception. What do we expect?

Reply:
a `ConnectionError` exception is now catched in case of connections issues to FortiGate. I there is problems the API Token, unfortunately fortiosapi does not raise a proper exception and the general exception is required to catch this situation.